cache/account-cache.js

```
@@ -0,0 +1,88 @@
+/* eslint-disable no-useless-constructor */
+const { default: axios } = require("axios");
+const moment = require("moment");
+const { CONFIG_BACKEND_URL } = require("../util/customTransforrmationsStore");
+const BaseCache = require("./base");
+const logger = require("../logger");
+
+class AccountCache extends BaseCache {
+  constructor() {
+    super();
+  }
+
+  static formCacheKey(accountId, workspaceId) {
+    return `${accountId}|${workspaceId}`;
+  }
+
+  static getTokenUrl(key) {
+    const [accountId, workspaceId] = key.split("|");
+    return `${CONFIG_BACKEND_URL}/dest/workspaces/${workspaceId}/accounts/${accountId}/token`;
+  }
+
+  hasTokenExpired(key) {
+    const tokenInfo = this.get(key);
+    return (
+      tokenInfo &&
+      tokenInfo.expirationDate &&
+      moment(tokenInfo.expirationDate).valueOf() < moment().valueOf()
+    );
+  }
+
+  static getTtl(tokenInfo) {
+    if (tokenInfo && tokenInfo.expirationDate) {
+      return (
+        Math.abs(
+          moment().valueOf() - moment(tokenInfo.expirationDate).valueOf()
+        ) / 1000
+      );
+    }
+    return 0;
+  }
+
+  async getToken(key) {
+    const tokenUrl = this.constructor.getTokenUrl(key);
+    const { data: secret } = await axios.post(tokenUrl);
+    return {
+      accessToken: secret.accessToken,
+      expirationDate: secret.expirationDate
+    };
+  }
+
+  async onExpired(k, v) {
+    try {
+      // Only AccessToken is being fetched in this call
+      const tokenInfo = await this.getToken(k);
+      this.set(k, tokenInfo);
+    } catch (error) {
+      logger.error(error);
```

We can throw the CustomError and let it propagate to version router catch block